lib/ansible/modules/database/postgresql/postgresql_copy.py

```diff
@@ -0,0 +1,427 @@
+  db:
+    description:
+    - Name of database to connect to.
+    type: str
+    aliases: [ login_db ]
+  session_role:
+    description:
+    - Switch to session_role after connecting.
+      The specified session_role must be a role that the current login_user is a member of.
+    - Permissions checking for SQL commands is carried out as though
+      the session_role were the one that had logged in originally.
+    type: str
+
+notes:
+- Supports PostgreSQL version 9.4+.
+- Transaction mode is not suitable for checking because of there may be
+  a lot of data and their actual upload can affect database performance.
+  So, in check mode, we just check the src/dst table availability
+  and return the query that aclually has not been executed.
+
+author:
+- Andrew Klychkov (@Andersson007)
+
+extends_documentation_fragment: postgres
+'''
+
+EXAMPLES = r'''
+- name: Copy text TAB-separated data from file /tmp/data.txt to acme table
+  postgresql_copy:
+    copy_from: /tmp/data.txt
+    dst: acme
+
+- name: Copy CSV (comma-separated) data from file /tmp/data.csv to columns id, name of table acme
+  postgresql_copy:
+    copy_from: /tmp/data.csv
+    dst: acme
+    columns: id,name
+    options:
+      format: csv
+
+- name: >
+    Copy text vertical-bar-separated data from file /tmp/data.txt to bar table.
+    The NULL values are specified as N
+  postgresql_copy:
+    copy_from: /tmp/data.csv
+    dst: bar
+    options:
+      delimiter: '|'
+      null: 'N'
+
+- name: Copy data from acme table to file /tmp/data.txt in text format, TAB-separated
+  postgresql_copy:
+    src: acme
+    copy_to: /tmp/data.txt
+
+- name: Copy data from SELECT query to/tmp/data.csv in CSV format
+  postgresql_copy:
+    src: 'SELECT * FROM acme'
+    copy_to: /tmp/data.csv
+    options:
+      format: csv
+
+- name: Copy CSV data from my_table to gzip
+  postgresql_copy:
+    src: my_table
+    copy_to: 'gzip > /tmp/data.csv.gz'
+    program: yes
+    options:
+      format: csv
+
+- name: >
+    Copy data from columns id, name of table bar to /tmp/data.txt.
+    Output format is text, vertical-bar-separated, NULL as N
+  postgresql_copy:
+    src: bar
+    columns:
+    - id
+    - name
+    copy_to: /tmp/data.csv
+    options:
+      delimiter: '|'
+      null: 'N'
+'''
+
+RETURN = r'''
+queries:
+  description: List of executed queries.
+  returned: always
+  type: str
+  sample: [ "COPY test_table FROM '/tmp/data_file.txt' (FORMAT csv, DELIMITER ',', NULL 'NULL')" ]
+src:
+  description: Data source.
+  returned: always
+  type: str
+  sample: "mytable"
+dst:
+  description: Data destination.
+  returned: always
+  type: str
+  sample: "/tmp/data.csv"
+'''
+
+try:
+    from psycopg2.extras import DictCursor
+except ImportError:
+    # psycopg2 is checked by connect_to_db()
+    # from ansible.module_utils.postgres
+    pass
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.database import SQLParseError, pg_quote_identifier
```

Comment:
`SQLParseError` - unused import?

Reply:
thanks, I pointed it out to @tcraxs too before, in his new module postgresql_sequence ;)